### PR TITLE
NIFI-6674 Expression language minus operation doesn't check for overflow

### PR DESCRIPTION
**Pull request by adarmiento**
#### Description of PR

*  `TestQuery` adds a unit test to cover the current behavior of `minus` overflow with `Long` values
*  Overflow issue is documented

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by adarmiento** [comment]
Hello @MikeThomsen 
Thanks for the update, I'll reply here also for #3739 and #3738

I paused working at this for this reason. I am not sure about whether it could be a clever idea to introduce a configuration value which, if enabled, makes it throw exceptions when overflow happens. 
In the meantime, I think it would be nice to merge docs and maybe the unit test confirming the behavior, in case in future we want to keep working at this. What do you think?

**Comment by MikeThomsen** [comment]
@adarmiento I think merging the docs makes sense, but not the unit tests because the behavior that's being tested is standard JVM behavior. You'd run into this behavior in any Java app that experiences overflow.

**Comment by adarmiento** [comment]
Ah ok, understood. 
Ok. Let's go for the doc only in this case

File: nifi-docs/src/main/asciidoc/expression-language-guide.adoc [hunk 1]
```diff
@@ -1838,7 +1838,7 @@ Divide. This is to preserve backwards compatibility and to not force rounding er
 [.function]
 === minus
 
-*Description*: [.description]#Subtracts a numeric value from the Subject.#
+*Description*: [.description]#Subtracts a numeric value from the Subject.# Does not provide handling for overflow.
```
Comment (mattyb149) [inline nifi-docs/src/main/asciidoc/expression-language-guide.adoc]: Shouldn't this be inside the `#` section?

Comment (adarmiento) [reply nifi-docs/src/main/asciidoc/expression-language-guide.adoc]: Yes it should. Nice catch

**Pull request closed without merging**